### PR TITLE
Polyfill the MaxAutoLockRenewalDuration

### PR DESCRIPTION
**Pull request by danielmarbach**
Follow up for #650 to make sure the old API supports setting this value too.

### CODE REVIEW

File: src/Transport/AzureServiceBusTransportSettingsExtensions.cs [hunk 1]
```diff
@@ -11,14 +11,17 @@
     /// </summary>
     public static partial class AzureServiceBusTransportSettingsExtensions
     {
+        const string Note = "As long as the persistence configuration API has not been adjusted to match the transport configuration API keep bumping the versions when working on a new major";
```
Comment (abparticular) [inline src/Transport/AzureServiceBusTransportSettingsExtensions.cs:14]: Is there an issue or documentation we can link to help Future Dev determine if the persistence configuration has been adjusted to match the transport configuration?

Comment (danielmarbach) [reply src/Transport/AzureServiceBusTransportSettingsExtensions.cs:14]: Unfortunately, at this stage we have only an internal tracking issue that we cannot link in the public documentation :(

**Comment by danielmarbach** [comment]
Apparently, it is not entirely clear whether we should or should not polyfill. I will be extracting the notes changes into a dedicated PR that we can merge independently of the shim API changes

**Comment by danielmarbach** [comment]
https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/664

**Review by lailabougria — APPROVED** [review]
Given that in this PR, we're going the extra mile for the user experience to provide options for both flavors of the API. Given that there's no full agreement and the action points item regarding this topic, I think we can merge this as is. If the conclusion is that we shouldn't go the extra mile, we can still remove it later.

**Pull request merged**